src/session-pool.ts

```diff
@@ -497,6 +499,10 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     this.database = database;
     this.options = Object.assign({}, DEFAULTS, options);
     this.options.min = Math.min(this.options.min!, this.options.max!);
+    this.options.databaseRole =
+      this.options.databaseRole !== null
```

Comment:
Should this not be just `this.options.databaseRole` instead of `this.options.databaseRole !== null` so that we don't let for example `undefined` override what is set in `database.databaseRole`.